### PR TITLE
[ROCm] Fixing build for CIs

### PR DESCRIPTION
**Pull request by ytaous**
**Description**: 
- fix CI for Windows GPU(cuda10.2) Validation Pipeline
- fix CI for Python Package Pipeline (Training for Rocm)

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/providers/cuda/cudnn_common.cc [hunk 1]
```diff
@@ -137,15 +137,9 @@ cudnnDataType_t CudnnTensor::GetDataType<half>() {
 }
 
 template <>
-
 cudnnDataType_t CudnnTensor::GetDataType<BFloat16>() {
-#if CUDNN_VERSION >= 8100
-  return CUDNN_DATA_BFLOAT16;
-#else
   ORT_THROW("cuDNN version is too low to support BFloat16.");
```
Comment (iK1D) [inline onnxruntime/core/providers/cuda/cudnn_common.cc:141]: May need to change this error msg, it's no longer version ralated.

File: onnxruntime/core/providers/rocm/miopen_common.cc [hunk 2]
```diff
@@ -93,7 +93,8 @@ miopenDataType_t MiopenTensor::GetDataType<half>() {
 
 template <>
 miopenDataType_t MiopenTensor::GetDataType<BFloat16>() {
-  return miopenBFloat16;
+  ORT_THROW("miopen version is too low to support BFloat16.");
```
Comment (iK1D) [inline onnxruntime/core/providers/rocm/miopen_common.cc:96]: Same here. Error msg is not that accurate.

File: onnxruntime/test/providers/cpu/reduction/reduction_ops_test.cc [hunk 3]
```diff
@@ -1536,9 +1536,10 @@ TEST(ReductionOpTest, ReduceSumBFloat16) {
 
 // on CUDA - this UT, with axes {0,2}, will go thru cudnn lib only if ATenOp is not initialized
 // on ROCM - miopen call succeeded, but results in data error, thus follow the same logic done in cudnn for now
+//           4.2 doesn't run properly, thus enable the UT only above 4.3
 // TODO - try ROCm 4.5.2 and/or double check the source code on BFloat16 support 
-#if defined(USE_CUDA) || defined(USE_ROCM)
-TEST(ReductionOpTest, DISABLED_ReduceSumBFloat16_2) {
+#if defined(USE_CUDA) || (defined(USE_ROCM)  && ROCM_VERSION >= 40300)
```
Comment (iK1D) [inline onnxruntime/test/providers/cpu/reduction/reduction_ops_test.cc:1541]: What's the problem when using 4.2?

Comment (ytaous) [reply onnxruntime/test/providers/cpu/reduction/reduction_ops_test.cc:1541]: The calculation is way off, rel_error = 1 mean it's 100% different - likely means the returned value is 0 (expected to be 33, and 45). We can look deeper if we want. But given our priority is ROCm 4.5, not sure we want to spend more time on ROC m 4.2. Thought?

[ RUN      ] ReductionOpTest.ReduceSumBFloat16_2
/onnxruntime_src/onnxruntime/test/providers/provider_test_utils.cc:345: Failure
The difference between 0 and rel_error is 1, which exceeds threshold, where
0 evaluates to 0,
rel_error evaluates to 1, and
threshold evaluates to 0.05000000074505806.
provider_type: ROCMExecutionProvider:
/onnxruntime_src/onnxruntime/test/providers/provider_test_utils.cc:345: Failure
The difference between 0 and rel_error is 1, which exceeds threshold, where
0 evaluates to 0,
rel_error evaluates to 1, and
threshold evaluates to 0.05000000074505806.
provider_type: ROCMExecutionProvider:
[  FAILED  ] ReductionOpTest.ReduceSumBFloat16_2 (31 ms)

**Review by iK1D — APPROVED** [review]

**Review by weixingzhang — APPROVED** [review]

**Comment by ytaous** [comment]
AMD failed as expected - should be fixed by https://github.com/microsoft/onnxruntime/pull/10574

**Pull request merged**